### PR TITLE
fix: pre-built binary support for Ubuntu 20.04

### LINKED ISSUE #2727: Goreleaser pre-built binaries don't work on Ubuntu 20.04
When ran locally on two different distributions of linux we hit

```
./celestia-appd: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by ./celestia-appd)
./celestia-appd: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by ./celestia-appd)
./celestia-appd: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by ./celestia-appd)
```

### PR DESCRIPTION
**Pull request by evan-forbes**
## Overview

I think this should fix #2727, I didn't see any docs to see how to format the version specified for ubuntu

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

## Testing

Tested with [celestia-app_Linux_x86_64.tar.gz](https://github.com/rootulp/celestia-app/releases/download/v1.0.0-rc7/celestia-app_Linux_x86_64.tar.gz) created on https://github.com/rootulp/celestia-app/releases/tag/v1.0.0-rc7

OS | Version | Works
--- | --- | ---
Ubuntu | 20.04 | ✅ 
Ubuntu | 23.04 | ✅

### CODE REVIEW

**Review by rootulp — APPROVED** [review]
LGTM but didn't try testing it

File: .goreleaser.yaml [hunk 1]
```diff
@@ -17,6 +17,7 @@ builds:
     binary: celestia-appd
     env:
       - SDKPath={{ "github.com/cosmos/cosmos-sdk/version" }}
+      - CGO_ENABLED=0
```
Comment (evan-forbes) [inline .goreleaser.yaml:20]: added this to also fix ledger support for macs for the prebuilt binaries

Comment (rootulp) [reply .goreleaser.yaml:20]: I expect we need to set 

```suggestion
      - CGO_ENABLED=1
```

because if CGO is disabled, Ledger won't work

Comment (rootulp) [reply .goreleaser.yaml:20]: Nvm this won't work https://github.com/celestiaorg/celestia-app/pull/2730

**Review by rootulp — APPROVED** [review]

**Review by staheri14 — APPROVED** [review]

**Pull request merged**